src/solver/infeasible-problem-analysis/watched-constraints.cpp

```diff
@@ -0,0 +1,238 @@
+    if (split_name.size() < 3)
+    {
+        err_msg += "constraint name '" + constraintName + "' misses '<' and/or '>' bracket";
+        throw StringIsNotWellFormated(err_msg);
+    }
+    if (split_name[1].empty())
+    {
+        err_msg += "constraint name '" + constraintName + "' must be of format '*<str>*'";
+        throw StringIsNotWellFormated(err_msg);
+    }
+    return split_name[1];
+}
+
+std::string timeStep(std::vector<std::string> splitName)
+{
+    return StringBetweenAngleBrackets(splitName.at(splitName.size() - 2));
+}
+
+std::string shortName(std::vector<std::string> splitName)
+{
+    return splitName.at(0);
+}
+
+std::string areaName(std::vector<std::string> splitName)
+{
+    return StringBetweenAngleBrackets(splitName.at(1));
+}
+
+std::string STSname(std::vector<std::string> splitName)
+{
+    return StringBetweenAngleBrackets(splitName.at(2));
+}
+
+namespace Antares::Optimization
+{
+
+// =======================================
+// Generic constraint logger
+// =======================================
+void WatchedConstraint::setConstraintName(std::string name)
+{
+    name_ = name;
+    boost::algorithm::split_regex(splitName_, name_, boost::regex("::"));
+}
+
+std::string WatchedConstraint::regexId()
+{
+    return regexId_;
+}
+
+// =======================================
+// Hourly BC logger
+// =======================================
+HourlyBC::HourlyBC()
+{
+    regexId_ = "::hourly::";
+}
+
+std::shared_ptr<WatchedConstraint> HourlyBC::clone() const
+{
+    return std::make_shared<HourlyBC>(*this);
+}
+
+std::string HourlyBC::infeasisibity()
+{
+    return "Hourly BC '" + shortName(splitName_) + "' at hour " + timeStep(splitName_);
+}
+
+std::string HourlyBC::infeasisibityCause()
+{
+    return "* Hourly binding constraints.";
+}
+
+// =======================================
+// Daily BC logger
+// =======================================
+DailyBC::DailyBC()
+{
+    regexId_ = "::daily::";
+}
+
+std::shared_ptr<WatchedConstraint> DailyBC::clone() const
+{
+    return std::make_shared<DailyBC>(*this);
+}
+
+std::string DailyBC::infeasisibity()
+{
+    return "Daily BC '" + shortName(splitName_) + "' at day " + timeStep(splitName_);
+}
+
+std::string DailyBC::infeasisibityCause()
+{
+    return "* Daily binding constraints,";
+}
+
+// =======================================
+// Weekly BC constraint
+// =======================================
+WeeklyBC::WeeklyBC()
+{
+    regexId_ = "::weekly::";
+}
+
+std::shared_ptr<WatchedConstraint> WeeklyBC::clone() const
+{
+    return std::make_shared<WeeklyBC>(*this);
+}
+
+std::string WeeklyBC::infeasisibity()
+{
+    return "Weekly BC '" + shortName(splitName_);
+}
+
+std::string WeeklyBC::infeasisibityCause()
+{
+    return "* Weekly binding constraints.";
+}
+
+// =======================================
+// Fictitious load constraint
+// =======================================
+FictitiousLoad::FictitiousLoad()
+{
+    regexId_ = "^FictiveLoads::";
+}
```

Comment:
well i you insist, at least this could be in-class initialized

Reply:
I do agree

Reply:
How would you do that ?

Reply:
Actually, we're not sure `infeasibility` will be called. So there's no need to construct the string in case we need it later.

Reply:
watched constraints no longer have a **regexId_** attribute : it's not associated to the class with in a factory.